dsl_parser/tests/test_get_attribute.py

```diff
@@ -98,30 +98,31 @@ class TestEvaluateFunctions(AbstractTestParser):
 
     def test_evaluate_functions(self):
 
-        def get_node_instances(node_id=None):
-            return [get_node_instance(node_id)]
-
-        def get_node_instance(node_instance_id):
-            result = NodeInstance({
-                'id': node_instance_id,
-                'node_id': 'webserver',
-                'runtime_properties': {
-                }
-            })
-            if node_instance_id == 'node1':
-                result.runtime_properties['a'] = 'a_val'
-            elif node_instance_id == 'node2':
-                result.runtime_properties['b'] = 'b_val'
-            elif node_instance_id == 'node3':
-                result.runtime_properties['c'] = 'c_val'
-            elif node_instance_id == 'node4':
-                result.runtime_properties['d'] = 'd_val'
-            return result
-
-        def get_node(node_id):
-            return Node({
-                'id': node_id,
-            })
+        class Storage(object):
+            def get_node_instances(self, node_id=None):
+                return [self.get_node_instance(node_id)]
+
+            def get_node_instance(self, node_instance_id):
+                result = NodeInstance({
+                    'id': node_instance_id,
+                    'node_id': 'webserver',
+                    'runtime_properties': {
+                    }
+                })
+                if node_instance_id == 'node1':
```

Comment:
what creating a dict for this

Reply:
refactored, however note that this wasn't added in this pr